darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,691 @@
+            """
+            Builds the kwargs dictionary for the transformation function.
+
+            Parameters
+            ----------
+            transformation
+                The transformation dictionary.
+            builtins
+                The built-in transformations read from the ForecastingWindowTransformer class.
+
+            Returns
+            -------
+            dict
+                The kwargs dictionary.
+            """
+            BUILTIN_TRANSFORMS = builtins
+            fn = transformation["function"]
+            useless_keys = ["function", "series_id", "comp_id"]
+            keys = list(transformation.keys() - useless_keys)
+
+            if fn in BUILTIN_TRANSFORMS:
+                # if builtin function, get the kwargs for the function group and the specific function
+
+                function_group_expected_args = set(
+                    BUILTIN_TRANSFORMS[fn][0].__code__.co_varnames
+                )
+                function_group_available_keys = list(
+                    function_group_expected_args.intersection(set(keys))
+                )
+                function_group_available_kwargs = {
+                    k: v
+                    for k, v in transformation.items()
+                    if k in function_group_available_keys
+                }
+
+                function_expected_args = set(
+                    getattr(
+                        getattr(pd.DataFrame(), BUILTIN_TRANSFORMS[fn][0].__name__)(
+                            **function_group_available_kwargs
+                        ),
+                        fn,
+                    ).__code__.co_varnames
+                )
+                function_available_keys = list(
+                    function_expected_args.intersection(set(keys))
+                )
+                function_available_kwargs = {
+                    k: v
+                    for k, v in transformation.items()
+                    if k in function_available_keys
+                }
+
+                return function_group_available_kwargs, function_available_kwargs
+            elif callable(fn):
+                # if user function, get the kwargs for the function
+                function_available_kwargs = {
+                    k: v for k, v in transformation.items() if k in keys
+                }
+                return function_available_kwargs
+            else:
+                raise_log(
+                    Exception("The transformation function is not valid."), logger
+                )
+
+        # get the kwargs from the transform() function call.
+        fillnamethod = kwargs.get(
+            "fillnamethod",
+            "bfill"
+            if (
+                ("navalue" not in kwargs)
+                or ("navalue" in kwargs and kwargs["navalue"] is None)
+            )
+            else None,
+        )
+        navalue = kwargs.get("navalue", None)
+
+        BUILTIN_TRANSFORMS = builtins
```

Comment:
Couldn't you maybe define the builtins outside of the class as a constant, in order to access them easily from everywhere?

Reply:
yep, can do